Tensile/Common.py

```diff
@@ -250,7 +250,7 @@
 
 # limitation for training
 globalParameters["MaxWorkspaceSize"] = 32 * 1024 * 1024 # max workspace for training (32M)
-globalParameters["MinKForGSU"] = 256 # min K size to use GlobalSplitU algorithm (only for HPA now)
+#globalParameters["MinKForGSU"] = 256 # min K size to use GlobalSplitU algorithm (only for HPA now)
```

Comment:
Is that only for HPA?
Maybe we need to delete this line.

Reply:
> Is that only for HPA? Maybe we need to delete this line.

Only for HPA.

Reply:
> > Is that only for HPA? Maybe we need to delete this line.
> 
> Only for HPA.

I do not think so.
I do not see any restriction to limit this parameter only for HPA case.
We should be able to use this for SGEMM, DGEMM, etc...

Reply:
I am not sure if it is only for HPA. I guess we can use this for all GEMM. We can test this though.

Reply:
> I am not sure if it is only for HPA. I guess we can use this for all GEMM. We can test this though.

As I pointed out in my other posts: I agreed with both you and Koji. That comment was copied and pasted from old comment. From the code, nothing prevents it from being used for other non-HPA GEMM. I will fix it.

Reply:
> > I am not sure if it is only for HPA. I guess we can use this for all GEMM. We can test this though.
> 
> As I pointed out in my other posts: I agreed with both you and Koji. That comment was copied and pasted from old comment. From the code, nothing prevents it from being used for other non-HPA GEMM. I will fix it.

I just realized that the old comment "only for HPA now" was because of the following condition in Contractions.py
  state['_GlobalAccumulation'] != None

We have already changed the definition of _GlobalAccumulation.
After the change, _GlobalAccumulation is None for HSS, BSS (means Dest=Compute).
It does not make sense we apply minKForGSU for HHS, but not for HSS.
Also, _GlobalAccumulation is not None if GSU algorithm is MultipleBuffer.
I do not see any good reason we apply minKForGSU for MultipleBuffer, but not for SingleBuffer.

Probably, we can apply minKForGSU for all data types.
We can simply check state['GlobalSplitU']>1 instead of checking _GlobalAccumulation.

You can add some test cases for SingleBuffer, non HPA.